cryostatio/cryostat
# feat(targets): dryrun query params for testing connectUrl and credentials

Related to https://github.com/cryostatio/cryostat-web/issues/592

This added the back-end support for test connection UI via `dryrun` query parameter. I chose another status code for JMX Auth Failure here is to avoid returning `427` that triggers openning up Auth Modal in the front-end. Another way is to re-use 427 but there must be a way for the front-end to know which requests are dryruns (i.e. maybe a custom header?)

**Manual Tests**

Using `localhost:9091` here to go around duplicate target error:

- With credentials
```bash
https --form POST https://localhost:8181/api/v2/targets?dryrun=true \
connectUrl='service:jmx:rmi:///jndi/rmi://localhost:9091/jmxrmi' \
alias="self" \
username="smoketest" \
password="smoketest" \
--verify=no \
--auth user:pass \
--auth-type basic
```

For non-dryrun requests (saving target), the workaround could be the front-end first posts a credentials with the matchExpression that matches this target. Then, cached when computing jvmid can grab it from there.

- Without credentials
```bash
https --form POST https://localhost:8181/api/v2/targets?dryrun=true \
connectUrl='service:jmx:rmi:///jndi/rmi://localhost:9091/jmxrmi' \
alias="self" \
--verify=no \
--auth user:pass \
--auth-type basic
```
